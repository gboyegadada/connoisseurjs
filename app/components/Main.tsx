import React, { ReactNode, FC } from 'react'
import { createSearchConnect } from '../redux/store'
import { SearchStatus } from '../types/search'
import CardList from './CardList'
import { MdSyncProblem } from 'react-icons/md'
import { Bounce } from './spinners/Bounce'

interface MainProps {
    children?: ReactNode
    className?: string
}

const SearchConnect = createSearchConnect({
    mapActions: actions => ({
        startSearch: actions.startSearch
    }),
    mapState: selectors => ({
        status: selectors.getStatus(),
        results: selectors.getSearchResults()
    })
})

const Main: FC<MainProps> = () => {

    return (
        <SearchConnect>
            {(_) => (
            <div className='content'>
                { _.status === SearchStatus.searching && <Bounce />  }
                { _.status === SearchStatus.complete && <CardList list={_.results} /> }
                { _.status === SearchStatus.error && (
                    <div className='flex-center'>
                        <h3>
                            <MdSyncProblem size={18} /> There was an error loading your search results.
                        </h3>
                    </div>
                 ) }
            </div>
        )}
        </SearchConnect>
    )
}

export default Main