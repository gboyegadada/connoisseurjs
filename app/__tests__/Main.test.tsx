import React from 'react';
import { shallow } from 'enzyme'

import Main from "../components/Main";


describe("<Main />", () => {
  test("should display a header, sidebar and main body", async () => {
    const main = shallow(<Main />)

    expect(main).toMatchSnapshot()
  });
});