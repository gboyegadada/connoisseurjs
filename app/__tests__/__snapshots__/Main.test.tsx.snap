// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should display a header, sidebar and main body 1`] = `
<ConnectAdvanced(reduxRenderPropRender)>
  <Component />
</ConnectAdvanced(reduxRenderPropRender)>
`;
